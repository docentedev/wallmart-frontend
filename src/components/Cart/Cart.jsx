import React from 'react'
import styles from './Cart.module.css'
import { X as CloseIcon } from 'react-feather';

import { currencyFormat, calcTotal } from '../../utils'

function Cart({ cartItems, removeItem, addItem, substractItem }) {
    const [show, setShow] = React.useState(false)

    React.useEffect(() => {
        if (cartItems.length > 0) setShow(true)
    }, [cartItems])

    return cartItems.length > 0 ? (
        <div className={`${styles.cart} ${show ? styles.cartShow : styles.cartHide}`}>
            {show ? (
                <div className={styles.cart__card} >
                    <div className={styles.cart__card__header}>
                        <button onClick={() => setShow(false)}>
                            <CloseIcon />
                        </button>
                    </div>
                    { cartItems.map((e, i) => (
                        <div key={i} >
                            {e.description} | {e.__quantity}<br />
                            {currencyFormat(e.price * e.__quantity)}
                            <button onClick={() => substractItem(e)}>-</button>
                            <button onClick={() => addItem(e)}>+</button>
                            <button onClick={() => removeItem(e)}>Quitar</button>
                        </div>
                    ))
                    }
                    < div >
                        {currencyFormat(calcTotal(cartItems))}
                    </div >
                </div >
            ) : (
                    <div className={styles.cart__button_open}>
                        <button onClick={() => setShow(true)}>Carrito</button>
                    </div>
                )}
        </div >
    ) : (<span />);
}

export default Cart;
