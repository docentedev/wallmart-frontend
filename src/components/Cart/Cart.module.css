.cart {
    position: fixed;
    z-index: 1;
    right: 0;
    top: 0;
    bottom: 0;
    margin: auto;
}

.cartShow {
    height: 80vh;
    max-height: 400px;
}

.cartHide {
    height: 80px;
}

.cart__card {
    background: white;
    box-shadow: 0 0 12px 0 #e6e5e5;
    height: 100%;
    width: 400px;
}

.cart_item_container {
    height: 220px;
    overflow-y: auto;
}