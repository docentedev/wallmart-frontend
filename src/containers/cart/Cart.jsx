import React from 'react'
import useCart from '../../hooks/useCart';

function Cart() {
    const { cartItems, addItem, substractItem, removeItem, getTotal } = useCart()

    return (
        <div>
            {cartItems.map((e, i) => (
                <div key={i} >
                    {e.description} | {e.__quantity}
                    <button onClick={() => substractItem(e)}>-</button>
                    <button onClick={() => addItem(e)}>+</button>
                    <button onClick={() => removeItem(e)}>Quitar</button>
                </div>
            ))}
            <div>
                {getTotal()}
            </div>
        </div>
    );
}

export default Cart;
