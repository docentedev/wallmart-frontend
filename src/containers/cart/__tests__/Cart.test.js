import React from 'react'
import TestRenderer from 'react-test-renderer'

import Cart from '../Cart'

jest.mock('react-router-dom', () => ({
    __esModule: true,
    Link: () => (<div className="link">Link</div>),
}))

jest.mock('react-redux', () => ({
    ...jest.requireActual('react-redux'),
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}))

jest.mock('../../../hooks/useCart', () => ({
    __esModule: true,
    default: () => ({
        get: (id) => id === 5 ? ({ __quantity: 2 }) : false,
        addItem: () => { },
        substractItem: () => { },
        cartItemsByBrand: {
            calcTotalByBrand: [],
        },
        cartItems: [],
    }),
}))

describe('Cart Test', () => {
    it('should render Cart without error', () => {
        const testRenderer = TestRenderer.create(<Cart />)
        // const testInstance = testRenderer.root

        expect(testRenderer.toJSON()).toMatchSnapshot()
    })
})
