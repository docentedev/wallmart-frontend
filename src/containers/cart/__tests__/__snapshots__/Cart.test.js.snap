// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Test should render Cart without error 1`] = `
<div
  className="cart_container"
>
  <div
    className="cart_container__no_product"
  >
    Aún no ha agregado ningún producto
    <div
      className="link"
    >
      Link
    </div>
  </div>
</div>
`;
